leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java

```diff
@@ -226,15 +226,15 @@ public LeshanServerBuilder setPrivateKey(PrivateKey privateKey) {
      * For RPK the public key will be extract from the first X509 certificate of the certificate chain. If you only need
      * RPK support, use {@link LeshanServerBuilder#setPublicKey(PublicKey)} instead.
      */
-    public LeshanServerBuilder setCertificateChain(X509Certificate[] certificateChain) {
+    public <T extends X509Certificate> LeshanServerBuilder setCertificateChain(T[] certificateChain) {
```

Comment:
pardon my ignorance but what is the difference between the two? What kind of certificates can I pass in now which I couldn't pass in before?

Reply:
I thought that arrays were invariant like generics but [there are not](https://stackoverflow.com/questions/2745265/is-listdog-a-subclass-of-listanimal-why-arent-javas-generics-implicitly-p/46496077#46496077).
So good question @AlexITC  can you provide an example where this modification helps ?

Reply:
oh, I forget to mention, I'm using Scala, Java arrays are handled as `Array<T>` collection, hence invariant.